### DISCUSSION TITLE
Are there calendar and timeline themes in doucusaurus?

### DISCUSSION THREAD

**Discussion by TheBestOrNothing** [facebook/docusaurus#4874] [Ideas]
I am trying to create my own calendar and timeline based on doucusaurus. So the theme of calendar and timeline is very helpful. Are there calendar and timeline themes in doucusaurus? Thanks.

**Comment by slorber** [comment]
I have no idea what you mean by calendar/timeline theme. 

We don't have that and you are the first to ask for it.

If you want us to consider adding this, please write an exhaustive feature request of how this should work exactly.

**Reply by HonkingGoose** [reply]
I'm not sure what kind of calender they want, so they'll have to explain that.

---

I think they want a timeline "component" that looks like this:

```
* **Docusaurus v2 in beta!!!**
|   We're out of alpha, and releasing on time, read the blog to learn more about this milestone.
|   Blabblabla
|
* **Docusaurus v2.0.0-alpha.75 released**
|   Breaking changes..
|   Blablabla
|
* **Timelines usually have a icon/graphic next to their headings
|  The rest of the timeline contains a line below the icon.
|  Timelines end after the last item in the timeline
```

Example of a pretty timeline: https://tailwindcss.com/docs search for the string 'Latest Updates' with your in-browser page search function.

**Reply by TheBestOrNothing** [reply]
Yes, that is what I mean. Thanks for your explanation.

**Comment by Siemienik** [comment]
Hi @TheBestOrNothing and @HonkingGoose.

AFAIK, Docuausus hasn't that possibility. However, It is easy to write it.

That what you need might be done by writing your own Remark Plugin. I did that someday ago for executing and presenting code snippets results. It looks like this one: 
https://github.com/handsontable/handsontable/commit/f7a96a948f805437c48c6811b1637db6393f1172#diff-b3fb42d668b3afce0646ae1baaea07b3378450a639bc984fb8b1aaaa87139a0f

More about writing remark plugins (and their available list) you can find on: https://github.com/remarkjs/remark/blob/main/doc/plugins.md

If you need some help or want to outsource it, contact me privately.

**Reply by HonkingGoose** [reply]
@TheBestOrNothing Did you see the response that @Siemienik wrote? Is that helpful for you?

**Reply by TheBestOrNothing** [reply]
@Siemienik @HonkingGoose  Thanks for your reply. Remark Plugin may be helpful to add timeline and calendar to docusaursu. Is there any code example to add remark plugin to docusaurus? Thanks.
